btree.go

```diff
@@ -64,6 +64,51 @@ type Item interface {
 	Less(than Item) bool
 }
 
+// FreeList represents a free list of btree nodes. By default each
+// BTree has its own FreeList, but multiple BTrees can share the same
+// FreeList.
+type FreeList struct {
+	freelist []*node
+	grow     bool
+}
+
+func NewFreeList() *FreeList {
+	return newFreeList(true)
+}
+
+func newDefaultFreeList() *FreeList {
+	return newFreeList(false)
+}
+
+func newFreeList(grow bool) *FreeList {
```

Comment:
You have a good point, since I am merely moving items from tree1 to tree2, one at a time maybe I don't need an unbounded freelist. It would be easy for me to run an experiment to see just how big my unbounded freelist actually gets. Since I am just moving items around rather than deleting items, I don't expect that my freelist would get very big.

However, for our application, I am more comfortable with letting our freelist grow as needed than to unknowingly allocate new btree nodes.  Just curious, how did you decide on 32 for the freelist size? What if we had a method where user could specify free list size and pass 0 for unbounded? would that be o.k?